client/js/components/GameView/MoveChoiceOverlay.vue

```diff
@@ -70,166 +70,175 @@ export default {
           break;
         case 2:
         case 9:
-          let oneOffDisabled = !this.isPlayersTurn;
-          let oneOffDisabledExplanation = "It's not your turn";
-          if (this.isPlayersTurn) {
-            if (this.opponentQueenCount >= 2) {
-              oneOffDisabled = true;
-              oneOffDisabledExplanation = `You can't play a ${cardRank} while your opponent has two or more queens`;
-            } else {
-              let validTargetExists;
-              // Twos
-              if (cardRank === 2) {
-                const numOpFaceCards = this.$store.getters.opponent.faceCards.length;
-                const numOpJacks = this.$store.getters.opponent.points.reduce(
-                  (jackCount, pointCard) => {
-                    return jackCount + pointCard.attachments.length;
-                  },
-                  0
-                );
-                const numTotalTargets = numOpFaceCards + numOpJacks;
-                validTargetExists = numTotalTargets >= 1;
-                if (allMovesAreDisabled) {
-                  oneOffDisabledExplanation = allMovesDisabledExplanation;
-                } else if (!validTargetExists) {
-                  oneOffDisabled = true;
-                  oneOffDisabledExplanation = 'There are no Royals to target';
-                }
-                // Nines
-              } else {
-                const numValidTargets =
-                  this.$store.getters.opponent.points.length +
-                  this.$store.getters.opponent.faceCards.length;
-                if (allMovesAreDisabled) {
-                  oneOffDisabledExplanation = allMovesDisabledExplanation;
-                } else if (numValidTargets === 0) {
-                  oneOffDisabled = true;
-                  oneOffDisabledExplanation = 'There are no point cards or Royals to target';
-                }
-              }
-            }
-          }
-          res = [
-            pointsMove,
-            scuttleMove,
-            {
-              displayName: 'One-Off',
-              eventName: 'targetedOneOff',
-              moveDescription: this.selectedCard.ruleText,
-              disabled: oneOffDisabled || allMovesAreDisabled,
-              disabledExplanation: oneOffDisabledExplanation,
-            },
-          ];
+          res = [this.pointsMove, this.scuttleMove, this.oneOffTwoAndNineMove];
           break;
         case 8:
           res = [
-            pointsMove,
-            scuttleMove,
+            this.pointsMove,
+            this.scuttleMove,
             // Glasses
             {
               displayName: 'Glasses',
               eventName: 'faceCard',
               moveDescription: 'Your opponent plays open handed',
-              disabled: allMovesAreDisabled,
-              disabledExplanation: disabledText,
+              disabled: this.allMovesAreDisabled,
+              disabledExplanation: this.disabledText,
             },
```

Comment:
I left the G8 one off object as is (same with queens and kings) since there is no logic being executed. I figured it might be cleaner to leave these in line since they're small, as opposed to creating two more computed properties.